packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,249 @@
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+`;
+
+const scrollContainer = css`
+  position: absolute;
+  top: 0;
+  right: 0;
+  left: 0;
+  bottom: 0;
+  overflow-y: auto;
+  animation: ${fadein} 150ms ease-in-out;
+`;
+
+const modalContentStyle = css`
+  transition: all 200ms ease-in-out;
+  margin: ${defaultSpacing}px auto;
+  padding: 36px;
+  color: ${uiColors.gray.dark3};
+  background-color: ${uiColors.white};
+  border-radius: 3px;
+  box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);
+  position: relative;
+  pointer-events: all;
+`;
+
+const modalSizes: { readonly [K in ModalSize]: string } = {
+  small: css`
+    width: 400px;
+  `,
+
+  default: css`
+    width: 720px;
+  `,
+
+  large: css`
+    ${mq({
+      width: ['720px', '720px', '960px'],
+    })}
+  `,
+};
+
+const closeButton = css`
+  color: ${uiColors.gray.dark1};
+  position: absolute;
+  cursor: pointer;
+  right: 12px;
+  top: 12px;
+`;
+
+const escapeKey = 27;
+
+interface ModalProps {
+  /**
+   * Content that will appear inside of the Modal component.
+   */
+  children: React.ReactNode;
+
+  /**
+   * Determines the active state of the modal
+   *
+   * default: `false`
+   */
+  active: boolean;
+
+  /**
+   * Specifies the size of the Modal.
+   *
+   * default: `default`
+   */
+  size?: ModalSize;
+
+  /**
+   * Callback to change the active state of the Modal.
+   *
+   */
+  setActive?: (
+    active: boolean,
+  ) => void | React.Dispatch<SetStateAction<boolean>>;
+
+  /**
+   * Callback to determine whether or not Modal should close when user tries to close it.
+   *
+   */
+  shouldClose?: () => boolean;
+
+  /**
+   * className applied to root overlay div.
+   *
+   */
+  className?: string;
+}
+
+/**
+ * # Modal
+ *
+ *  Modals place content on top of main window.
+ *
+```
+<Modal
+  active
+  size="large"
+  setActive={setActive}
+  shouldClose={() => console.log('Modal is closing now!')}
+  >
+  Modal content!
+</Modal>
+```
+ * @param props.active Boolean to describe whether or not Modal is active.
+ * @param props.size String to determine size of Modal. ['small', 'default', 'large']
+ * @param props.setActive Callback to change the active state of Modal.
+ * @param props.children Content to appear inside of Modal container.
+ * @param props.shouldClose Callback to determine whether or not Modal should close when user tries to close it.
+ * @param props.className className applied to overlay div.
+ *
+ */
+function Modal({
+  active = false,
+  size = ModalSize.Default,
+  setActive,
+  children,
+  shouldClose,
+  className,
+  ...rest
+}: ModalProps) {
+  if (!active) {
+    return null;
+  }
+
+  const scrollContainerRef = useRef<HTMLDivElement>(null);
+
+  const handleClose = useCallback(() => {
+    // Don't close modal if shouldClose returns false or no setActive callback is passed
+    if (!setActive || (shouldClose && !shouldClose())) {
+      return;
+    }
+
+    setActive(false);
+  }, [setActive, shouldClose]);
```

Comment:
If a user doesn't wrap these functions with `useCallback` they're always going to be different on every run, which may reduce the value of using `useCallback` to begin with. May be fine to skip `useCallback` here...or keep it. Not sure what the real difference is going to be.

Reply:
I think we did it because in order for the useCallback in handleEscape to work correctly we'd have to include setActive in the list of dependencies, but it made more semantic sense to break it out the way that we did. (i.e. handleEscape includes handleClose and handleClose includes setActive) cc: @jetpacmonkey

Reply:
Yeah the chain of dependencies was easier to follow this way. Having `setActive` as a dependency in a function that doesn't directly reference `setActive` seemed likely to be confusing.